src/uu/split/src/number.rs

```diff
@@ -0,0 +1,513 @@
+/// to `vec![radix - 1, 0]`, it instead increases its width by one and
+/// resets its value to 0. For example, if the radix were 3, the
+/// digits were `vec![1, 2]`, and we called [`increment`], then the
+/// digits would become `vec![0, 0, 0]`. In this way, the width grows
+/// by one each time the most significant digit would have achieved
+/// its maximum value.
+///
+/// This notion of "incrementing" here does not match the notion of
+/// incrementing the *value* of the number, it is just an abstract way
+/// of updating the representation of the number in a way that is only
+/// useful for the purposes of the `split` program.
+///
+/// # Displaying
+///
+/// This number is only displayable if `radix` is 10 or `radix` is
+/// 26. If `radix` is 10, then the digits are concatenated and
+/// displayed as a fixed-width decimal number with a prefix of `n - 2`
+/// instances of the character '9', where `n` is the number of digits.
+/// If `radix` is 26, then each digit is translated to the
+/// corresponding lowercase ASCII alphabetic character (that is, 'a',
+/// 'b', 'c', etc.) and concatenated with a prefix of `n - 2`
+/// instances of the character 'z'.
+///
+/// This notion of displaying the number is specific to the `split`
+/// program.
+#[derive(Clone)]
+pub struct DynamicWidthNumber {
+    radix: u8,
+    digits: Vec<u8>,
+}
+
+impl DynamicWidthNumber {
+    /// Instantiate a number of the given radix, starting with width 2.
+    ///
+    /// This associated function returns a new instance of the struct
+    /// with the given radix and a width of two digits, both 0.
+    pub fn new(radix: u8) -> DynamicWidthNumber {
+        DynamicWidthNumber {
+            radix,
+            digits: vec![0, 0],
+        }
+    }
+
+    /// Set all digits to zero.
+    fn reset(&mut self) {
+        for i in 0..self.digits.len() {
+            self.digits[i] = 0;
+        }
+    }
+
+    /// Increment this number.
+    ///
+    /// This method adds one to this number. The first time that the
+    /// most significant digit would achieve its highest possible
+    /// value (that is, `radix - 1`), then all the digits get reset to
+    /// 0 and the number of digits increases by one.
+    ///
+    /// This method never returns an error.
+    fn increment(&mut self) -> Result<(), Overflow> {
+        for i in (0..self.digits.len()).rev() {
+            // Increment the current digit.
+            self.digits[i] += 1;
+
+            // If the digit overflows, then set it to 0 and continue
+            // to the next iteration to increment the next most
+            // significant digit. Otherwise, terminate the loop, since
+            // there will be no further changes to any higher order
+            // digits.
+            if self.digits[i] == self.radix {
+                self.digits[i] = 0;
+            } else {
+                break;
+            }
+        }
+
+        // If the most significant digit is at its maximum value, then
```

Comment:
For `DynamicWidthNumber`, an overflow results in resetting the digits to zero and increasing the number of digits by one.